x/ecocredit/migrations/v3/core.go

```diff
@@ -0,0 +1,385 @@
+			}
+		}
+
+		// delete class info from old store
+		if err := classInfoTable.Delete(sdkCtx, &ClassInfo{ClassId: classInfo.ClassId}); err != nil {
+			return err
+		}
+	}
+
+	// migrate credit type sequence to ORM v1
+	cItr, err := creditTypeSeqTable.PrefixScan(sdkCtx, nil, nil)
+	if err != nil {
+		return err
+	}
+	defer cItr.Close()
+
+	for {
+		var ctype CreditTypeSeq
+		if _, err := cItr.LoadNext(&ctype); err != nil {
+			if orm.ErrIteratorDone.Is(err) {
+				break
+			}
+
+			return err
+		}
+		if err := ss.ClassSequenceTable().Save(ctx, &api.ClassSequence{
+			CreditType:  ctype.Abbreviation,
+			NextClassId: ctype.SeqNumber,
+		}); err != nil {
+			return err
+		}
+
+		// delete credit type sequence from old store
+		if err := creditTypeSeqTable.Delete(sdkCtx, &ctype); err != nil {
+			return err
+		}
+	}
+
+	// migrate projects
+	// TODO: update with actual data
+	projects := []*api.ProjectInfo{
+		{
+			Name:            "P01",
+			ClassId:         1,
+			Admin:           sdk.AccAddress("cosmos154hmhstk3gpkv2ndec8zkjkc5c3svutcqcswne"),
+			ProjectLocation: "AB-CDE FG1 345",
+			Metadata:        "metadata",
+		},
+	}
+
+	projectSeqMap := make(map[uint64]uint64) // map of a credit classID to project sequence
+	for _, p := range projects {
+		dest := api.ProjectInfo{
+			Name:            p.Name,
+			Admin:           p.Admin,
+			ClassId:         p.ClassId,
+			ProjectLocation: p.ProjectLocation,
+			Metadata:        p.Metadata,
+		}
+		pID, err := ss.ProjectInfoTable().InsertReturningID(ctx, &dest)
+		if err != nil {
+			return err
+		}
+
+		projectIDsMap[classIDsMap[p.ClassId]] = pID
+		if v, ok := projectSeqMap[p.ClassId]; ok {
+			projectSeqMap[p.ClassId] = v + 1
+		} else {
+			projectSeqMap[p.ClassId] = 2
+		}
+
+	}
+
+	// add project sequence
+	keys := make([]int, 0, len(projectSeqMap))
+	for k := range projectSeqMap {
+		keys = append(keys, int(k))
+	}
+	sort.Ints(keys)
+
+	for _, k := range keys {
+		key := uint64(k)
+		err = ss.ProjectSequenceTable().Save(ctx, &api.ProjectSequence{
+			ClassId:       key,
+			NextProjectId: projectSeqMap[key],
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	// migrate credit batches to ORM v1
+	batchIDsMap := make(map[string]batchMapT) // map of a batch denom to batch-id and amount cancelled
+	batchSeqMap := make(map[uint64]uint64)    // map of a project-id to batch sequence
+	batchItr, err := batchInfoTable.PrefixScan(sdkCtx, nil, nil)
+	if err != nil {
+		return err
+	}
+	defer batchItr.Close()
+
+	for {
+		var batchInfo BatchInfo
+		if _, err := batchItr.LoadNext(&batchInfo); err != nil {
+			if orm.ErrIteratorDone.Is(err) {
+				break
+			}
+			return err
+		}
+
+		bInfo := api.BatchInfo{
+			ProjectId:    projectIDsMap[batchInfo.ClassId],
+			BatchDenom:   batchInfo.BatchDenom,
+			Metadata:     string(batchInfo.Metadata),
+			StartDate:    timestamppb.New(*batchInfo.StartDate),
+			EndDate:      timestamppb.New(*batchInfo.EndDate),
+			IssuanceDate: nil, // TODO: add issuance date
```

Comment:
Is there something blocking this todo?

Reply:
Oof... Yea, not sure how we should do this. We don't want this to be user controlled and we can't query the event, so we would have to manually update it (...and we would need to do redwood separately). Adding this to #980.

Reply:
Oof... Yea, this will probably need to be done manually along with the curator address for baskets. We don't want this to be user controlled and we can't query the event. Added this item to #980.

Reply:
Yes, This needs to be done manually. Can we leave it empty for existing credit batches?